paimon-common/src/main/java/org/apache/paimon/fileindex/FileIndexer.java

```diff
@@ -18,25 +18,56 @@
 
 package org.apache.paimon.fileindex;
 
-import org.apache.paimon.fileindex.bloomfilter.BloomFilterFileIndex;
 import org.apache.paimon.options.Options;
 import org.apache.paimon.types.DataType;
 
-import static org.apache.paimon.fileindex.bloomfilter.BloomFilterFileIndex.BLOOM_FILTER;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.ServiceLoader;
 
 /** File index interface. To build a file index. */
 public interface FileIndexer {
 
+    Logger LOG = LoggerFactory.getLogger(FileIndexer.class);
+
     FileIndexWriter createWriter();
 
     FileIndexReader createReader(byte[] serializedBytes);
 
     static FileIndexer create(String type, DataType dataType, Options options) {
-        switch (type) {
-            case BLOOM_FILTER:
-                return new BloomFilterFileIndex(dataType, options);
-            default:
-                throw new RuntimeException("Doesn't support filter type: " + type);
+
+        FileIndexerFactory fileIndexerFactory = FileIndexerLoadFactory.load(type);
+        return fileIndexerFactory.create(dataType, options);
+    }
+
+    /** Load factory to load FileIndexerFactory. */
+    class FileIndexerLoadFactory {
```

Comment:
We don't need to use this class FileIndexLoadFactory, it doesn't make much sense. Static variables and methods are directly placed in FileIndexer

Reply:
You can just create `FileIndexerFactoryUtils`.